pypiserver/core.py

```diff
@@ -181,7 +181,8 @@ def guess_pkgname_and_version(path):
 
 
 def normalize_pkgname(name):
-    return name.lower().replace("-", "_")
+    """Perform PEP 503 normalization"""
+    return re.sub(r"[-_.]+", "-", name).lower()
```

Comment:
Suggestion: use `pkg_resources.safe_name().lower()` instead, since `pkg_resources` should stay up to date with any future PEP changes